app/api_topologies.go

```diff
@@ -80,22 +80,27 @@ func updateFilters(rpt report.Report, topologies []APITopologyDesc) []APITopolog
 		ns = append(ns, namespace)
 	}
 	sort.Strings(ns)
-	for i, t := range topologies {
-		if t.id == podsID || t.id == servicesID || t.id == deploymentsID || t.id == replicaSetsID {
-			topologies[i] = updateTopologyFilters(t, []APITopologyOptionGroup{
-				kubernetesFilters(ns...), unmanagedFilter,
-			})
+	if len(ns) > 0 { // We only want to apply k8s filters when we have k8s-related nodes
```

Comment:
I'd invert that, returning `topologies` in the branch, make it operate on `len(namespace)` and move it before all this `ns` stuff.